### ISSUE TITLE
Deprecated `/` for division use in SASS themes

### ISSUE THREAD

**Issue by ceisele-r** [telerik/kendo-themes#2505]
**Describe the bug**
Throughout `kendo-theme-default`, `/` is being used for division in SASS code that is deprecated by SASS lang.
See https://sass-lang.com/d/slash-div

**To reproduce**

1. Use scss `kendo-theme-default` in project
2. Compile using `sass: 1.34.0`

**Expected behavior**
No deprecation warnings

**Screenshots**
```
DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($padding-x, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
73 │ $padding-x-sm: $padding-x / 2 !default;
   │                ^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 73:16  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($padding-y, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
74 │ $padding-y-sm: $padding-y / 2 !default;
   │                ^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 74:16  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($border-radius, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
80 │ $border-radius-sm: $border-radius / 2 !default;
   │                    ^^^^^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 80:20  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div(20, 14)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
307 │ $line-height: ( 20 / 14 ) !default;
    │                 ^^^^^^^
    ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 307:17  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9    @import
    src\common\kendo.scss 6:9                                               root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($drop-hint-arrow-size, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
315 │ $drop-hint-arrow-spacing: ($drop-hint-arrow-size / 2) !default;
    │                            ^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 315:28  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9    @import
    src\common\kendo.scss 6:9                                               root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($padding-x, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
73 │ $padding-x-sm: $padding-x / 2 !default;
   │                ^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 73:16  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($padding-y, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
74 │ $padding-y-sm: $padding-y / 2 !default;
   │                ^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 74:16  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($border-radius, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

   ╷
80 │ $border-radius-sm: $border-radius / 2 !default;
   │                    ^^^^^^^^^^^^^^^^^^
   ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 80:20  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9   @import
    src\common\kendo.scss 6:9                                              root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div(20, 14)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
307 │ $line-height: ( 20 / 14 ) !default;
    │                 ^^^^^^^
    ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 307:17  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9    @import
    src\common\kendo.scss 6:9                                               root stylesheet

DEPRECATION WARNING: Using / for division is deprecated and will be removed in Dart Sass 2.0.0.

Recommendation: math.div($drop-hint-arrow-size, 2)

More info and automated migrator: https://sass-lang.com/d/slash-div

    ╷
315 │ $drop-hint-arrow-spacing: ($drop-hint-arrow-size / 2) !default;
    │                            ^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
    node_modules\@progress\kendo-theme-default\scss\_variables.scss 315:28  @import
    node_modules\@progress\kendo-theme-default\scss\grid\_index.scss 1:9    @import
    src\common\kendo.scss 6:9                                               root stylesheet
```

**Affected package (please remove the unneeded items)**
Probably affects all but since we only use `theme-default`, I cannot for sure tell for the others.
* theme-default
* theme-bootstrap
* theme-material
* theme-tasks

**Affected suites (please remove the unneeded items)**
* Kendo UI for jQuery
* Kendo UI for React

Probably also the others but I am also unsure about this.

**Affected browsers (please remove the unneeded items)**
* Not Applicable, happens during SASS compile

**Build system information (please remove the unneeded items)**
* OS: All
* Node version: 14
* NPM version: 6

**Additional context**
Add any other context about the bug.

**Comment by joneff** [comment]
Just to be clear on something: **this is not a bug**. And here is why:

* this is a warning and not an error -- the compilation succeeds
  * it's a warning starting with dart-sass 1.33.0 (you are using 1.34.0, so you see them)
  * this is a warning for upcoming changes in dart-sass compiler v2.0.0 which is going to be breaking change
* you can suppressing the warnings

Let's go over each item individually:

**warning, not an error**

Indeed. The compilation succeeds. Yes, dart-sass is nagging about it and kinda giving heads up for breaking change in v2, but we already know v2 is going to be breaking. In all fairness, even minor versions are sometimes breaking

**you can suppress the warnings**

There is a `quietDeps` flag in sass. Depending how you compile the themes (cli or webpack) it's applied differently. Here is the syntax for webpack:

```js
{
  loader: require.resolve('sass-loader'),
  options: {
    sassOptions: { quietDeps: true },
  },
}
```

For cli you can use `--quiet`.

**All that being said**

When sass v2 is out, we are going to make a v2 compatible version. Until then, however, warnings like this are inevitable...

**Comment by ceisele-r** [comment]
@joneff totally agree, it's currently not a bug.
This was meant to raise the "issue" of the current implementation (=use of `/` for division) that will make it break with sass v2.

**Comment by joneff** [comment]
Usage of / for division is the least of our problems, when it comes to dart-sass v2.

The bigger issue at hand is the changed syntax and the performance:
* syntax is backwards incompatible with node-sass
  * supporting the new syntax requires a big... no that's not it ... humongous refactoring of the project
* node-sass will no longer be updated (it will never understand the new syntax)
* there is significant performance penalty associated with dart-sass.

Actually, we don't really see the complete rewrite of the themes as an issue, nor the missing backwards compatibility -- we've chosen sass and if this is the future of sass, so be it.

What we are concerned about is the compilation time. Sure, customers rarely compile, but when they do ... on my local machine with all the warnings it takes about two full minutes to compile a theme with dart-sass. If i suppress the warnings, it's less, but still considerable amount.

Build time affect's us a lot!

Lastly, our third party dependencies, like Bootstrap limit the extent in which we can migrate -- if Bootstrap do not migrate to the new syntax, we can't migrate our bootstrap theme...

We are aware of the issues / differences in syntax and are not what you call eager for dart-sass v2 to come by.

**Comment by ceisele-r** [comment]
Thanks for the background info @joneff ! And I totally agree that build time is an important piece.

**Comment by ceisele-r** [comment]
@joneff just a short info as you mentioned bootstrap - they actually implemented a solution already in `v5.0.2`: https://blog.getbootstrap.com/2021/06/22/bootstrap-5-0-2/

**Comment by joneff** [comment]
Should we decide to implement a workaround, I think we'll go more into how things are handled in material https://github.com/angular/components/commit/a4043f41f539ed910c80c780e1815f5625282b94#diff-4b0d3692fae1e26970f5365a78db4ff81acb362f79ef12252081df8b285d38e8R49. 

The bootstrap approach seem a bit too complicated -- https://github.com/twbs/bootstrap/commit/be17444756c48e3a892c36507f859cd841bd8c1f#diff-de6111a37eb67d027961043ce738e80a3ac4ad61a6e7704d35cb3168369cba01R223

**Comment by x2764tech** [comment]
> **you can suppress the warnings**

Yes, but this suppresses warning in my code as well, which isn't what I want.

**Comment by msmolka** [comment]
related #3029

other recommendation
Recommendation: math.div($spacer, 2) or calc($spacer / 2)

**Comment by Ristaaf** [comment]
Or you just do what the autofixer for sass does in most cases and changes the division into a multiplication, most divisions in kendo-themes are divisions by 2 which is easily changed to multiplications of 0.5 instead, so that $border-radius / 2 is replaced with $border-radius * 0.5 then only one I see in the errors are the 20/14 division which I guess can be replaced with an actual value too with  the required precision to be ok (2.857?). This would make it work both with dart-sass 2 and node-sass.

**Comment by joneff** [comment]
Resolved with #4175 -- we use our own `k-math-div` function.

Since we do have some ongoing development with branches no using the function, a nefarious slash might creep in. But we'll make sure to address it before the release!
